File: solutions/python/VB_T5_005_checked_index.py
"""VB-T5-005: Checked Array Index -- Python baseline."""
def safe_index(arr: list[int], n: int) -> int:
    if n >= len(arr): return -1
    return arr[n]

if __name__ == "__main__":
    assert safe_index([10,20,30], 1) == 20
    assert safe_index([10], 5) == -1
    print("VB-T5-005 ok")

File: solutions/typescript/VB_T5_005_checked_index.ts
// VB-T5-005: Checked Index -- TypeScript baseline
function safeIndex(arr: number[], n: number): number { return n >= arr.length ? -1 : arr[n]; }
console.assert(safeIndex([10,20,30], 1) === 20);
console.assert(safeIndex([10], 5) === -1);
